include/engine/guidance/step_maneuver.hpp

```diff
@@ -0,0 +1,26 @@
+#ifndef ENGINE_GUIDANCE_STEP_MANEUVER_HPP
+#define ENGINE_GUIDANCE_STEP_MANEUVER_HPP
+
+#include "util/coordinate.hpp"
+#include "extractor/turn_instructions.hpp"
+
+namespace osrm
+{
+namespace engine
+{
+namespace guidance
+{
+
+struct StepManeuver
+{
+    util::FixedPointCoordinate location;
+    double heading_before;
+    double heading_after;
+    extractor::TurnInstruction instruction;
+};
```

Comment:
This only allows a single name, does it? I was thinking about something like `take exit **number** towards **direction**`. Is there a way do give these additional hints or am I supposed to encode these into some of the fields? I guess way_name would be the exit number (as a name for the ramp?), but where would I add the additional hints for the direction name?

Using the name could be a possibility, but it would somewhat break the expected result if such hints are encoded into the name field.